### PR TITLE
Added Color and Image extensions

### PR DESCRIPTION
**Pull request by wolfenrain**
# Description

Follow up PR from https://github.com/flame-engine/flame/pull/646#discussion_r569535816, introducing an image extension and color extension.

I also added a few methods that I personally use, so if any of those are deemed unnecessary in the context of Flame then I am okay with removing them.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

If something is unclear, please submit the PR anyways and ask about what you thought was unclear.

- [x] This branch is based on the latest `master`
- [x] I have added an entry under `[next]` in `CHANGELOG.md`
- [x] I have formatted my code with `flutter format`
- [ ] I have made corresponding changes to the documentation
- [ ] I have added examples for new features in `doc/examples`
- [x] The continuous integration (CI) is passing

### CODE REVIEW

File: lib/src/extensions/color.dart
Comment (erickzanardo) [inline lib/src/extensions/color.dart:11]: Oh, this is excellent!

File: lib/src/extensions/image.dart
Comment (erickzanardo) [inline lib/src/extensions/image.dart:9]: Since you created this extension, there is a method that I think that would be cool to be here. We could have a `fromSprite` method, which would take a `Sprite` instance and return an `Image` instance which represents the "cropped" image that the sprite maps. What do you think?

Motivation: This would help on a new feature that I am planning on creating on fire atlas

**Review by erickzanardo — APPROVED** [review]
LGTM

File: lib/src/extensions/image.dart
Comment (erickzanardo) [reply lib/src/extensions/image.dart:9]: But in my use case, I would like to get the image from a Sprite, yeah, I could use composition, but this seems weird to me to use a class that compose an image, when I want to do the opposite operation. I don't think it would be such a redundant API since both classes have places which they fit better.

We could also put this on the Sprite class maybe, having a `toImage` method there.

**Review by spydon — APPROVED** [review]
Good stuff!

File: lib/src/extensions/image.dart
Comment (luanpotter) [reply lib/src/extensions/image.dart:9]: how about getCroppedImage() ? we can use ImageComposition underneath to impl it but I think having a method is useful!

File: lib/src/extensions/image.dart
Comment (luanpotter) [inline lib/src/extensions/image.dart:8]: I would add some comments here regarding:

this is *very* slow. do **not** do this every tick in your game, or your device, emulator, computer, city's power grid and the www will explode.

File: CHANGELOG.md
Comment (luanpotter) [inline CHANGELOG.md:11]: I think an example could be useful? but it can be a followup on the new repo instead of adding examples on the old structure

**Review by luanpotter — APPROVED** [review]
LGTM but I defer to @renancaraujo as the image manipulation expert

File: lib/src/extensions/image.dart
Comment (wolfenrain) [reply lib/src/extensions/image.dart:9]: I have discussed this with @erickzanardo  privately as well and we concluded a method on the `Sprite` class would be useful indeed, and I'll be following up on that after this PR is merged.

**Review by renancaraujo — APPROVED** [review]

**Pull request merged**
